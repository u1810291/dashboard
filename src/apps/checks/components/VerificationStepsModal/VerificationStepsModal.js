import PropTypes from 'prop-types';
import React, { useCallback, useState } from 'react';
import { useIntl } from 'react-intl';
import { Modal } from 'apps/overlay';
import { Box, Button } from '@material-ui/core';
import { CheckboxGroup } from 'apps/ui';

export function VerificationStepsModal({ items, onSave, values }) {
  const intl = useIntl();
  const [selectedOptions, setSelectedOptions] = useState(values);
  const verificationOptions = items.map((item) => ({
    label: intl.formatMessage({ id: `verification_items.${item}` }),
    value: item,
  }));
  const handleCheckboxGroupChange = useCallback((newValues) => {
    setSelectedOptions(newValues);
  }, [setSelectedOptions]);
  const handleSave = useCallback(() => {
    onSave(selectedOptions);
  }, [onSave, selectedOptions]);

  return (
    <Modal>
      <main>
        <Box mb={2}>
          <h1>
            {intl.formatMessage({ id: 'fragments.configuration.verification_steps_modal.title' })}
            <p>
              {intl.formatMessage({ id: 'fragments.configuration.verification_steps_modal.subtitle' })}
            </p>
          </h1>
        </Box>
        <CheckboxGroup
          items={verificationOptions}
          values={selectedOptions}
          onChange={handleCheckboxGroupChange}
        />
      </main>
      <footer>
        <Button
          variant="contained"
          color="primary"
          size="medium"
          disabled={!selectedOptions.length}
          onClick={handleSave}
        >
          {intl.formatMessage({ id: 'done' })}
        </Button>
      </footer>
    </Modal>
  );
}

VerificationStepsModal.propTypes = {
  items: PropTypes.arrayOf(PropTypes.string),
  onSave: PropTypes.func,
  values: PropTypes.arrayOf(PropTypes.string),
};

VerificationStepsModal.defaultProps = {
  items: [],
  onSave: () => {},
  values: [],
};
