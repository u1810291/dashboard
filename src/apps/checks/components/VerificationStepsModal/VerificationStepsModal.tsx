import React, { useCallback, useState, useMemo } from 'react';
import { useIntl } from 'react-intl';
import { Modal } from 'apps/overlay';
import { Box, Button } from '@material-ui/core';
import { CheckboxGroup } from 'apps/ui';
import { CustomDocumentResponse } from 'models/CustomDocument.model';

export function VerificationStepsModal({
  documents = [],
  customDocuments = [],
  onSave,
  values,
  mode = 'both',
}: {
  documents?: string[];
  customDocuments?: CustomDocumentResponse[];
  onSave: (selectedOptions: string[]) => void;
  values: string[];
  mode?: 'regular' | 'custom' | 'both';
}) {
  const intl = useIntl();
  const [selectedOptions, setSelectedOptions] = useState(values);
  const verificationOptions = useMemo(() => {
    const regularOptions = ['both', 'regular'].includes(mode) ? documents.map((item) => ({
      label: intl.formatMessage({ id: `verification_items.${item}` }),
      value: item,
    })) : [];
    const customOptions = ['both', 'custom'].includes(mode) ? customDocuments.map((item) => ({
      label: item.name,
      value: item.type,
    })) : [];

    return [
      ...regularOptions,
      ...customOptions,
    ];
  }, [documents, customDocuments, intl, mode]);
  const handleCheckboxGroupChange = useCallback((newValues) => {
    setSelectedOptions(newValues);
  }, [setSelectedOptions]);
  const handleSave = useCallback(() => {
    onSave(selectedOptions);
  }, [onSave, selectedOptions]);

  return (
    <Modal>
      <main>
        <Box mb={2}>
          <h1>
            {intl.formatMessage({ id: 'fragments.configuration.verification_steps_modal.title' })}
            <p>
              {intl.formatMessage({ id: 'fragments.configuration.verification_steps_modal.subtitle' })}
            </p>
          </h1>
        </Box>
        <CheckboxGroup
          items={verificationOptions}
          values={selectedOptions}
          onChange={handleCheckboxGroupChange}
        />
      </main>
      <footer>
        <Button
          variant="contained"
          color="primary"
          size="medium"
          disabled={!selectedOptions.length}
          onClick={handleSave}
        >
          {intl.formatMessage({ id: 'done' })}
        </Button>
      </footer>
    </Modal>
  );
}
