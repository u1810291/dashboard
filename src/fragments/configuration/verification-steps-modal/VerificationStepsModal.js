import React from 'react'
import { FormattedMessage } from 'react-intl'
import Button from 'src/components/button'
import { Modal } from 'src/components/modal'
import CheckboxGroup from 'src/components/checkbox-group'

export default class VerificationStepModal extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      values: props.values
    }
  }

  static defaultProps = {
    values: [],
    items: [],
    onSave: () => {}
  }

  render() {
    const { values } = this.state
    const items = this.props.items.map(item => ({
      label: <FormattedMessage id={`verification_items.${item}`} key={item} />,
      value: item
    }))
    const { onSave } = this.props
    return (
      <Modal>
        <main>
          <h1>
            <FormattedMessage id="fragments.configuration.verification_steps_modal.title" />
            <p>
              <FormattedMessage id="fragments.configuration.verification_steps_modal.subtitle" />
            </p>
          </h1>
          <CheckboxGroup
            items={items}
            values={values}
            onChange={values => this.setState({ values })}
          />
        </main>
        <footer className="mgi-items">
          <Button
            buttonStyle="primary"
            disabled={!this.state.values.length}
            onClick={() => onSave(values)}
          >
            <FormattedMessage id="done" />
          </Button>
        </footer>
      </Modal>
    )
  }
}
