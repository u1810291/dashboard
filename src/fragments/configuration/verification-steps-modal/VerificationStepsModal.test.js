import React from 'react'
import { shallow } from 'enzyme'
import VerificationStepsModal from '.'

describe('VerificationStepModal', () => {
  it('is disaled with empty values', () => {
    const props = {
      values: [],
      items: ['passport', 'national-id']
    }
    const component = shallow(<VerificationStepsModal {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('is enabled and has pre-checked values', () => {
    const props = {
      values: ['national-id'],
      items: ['passport', 'national-id']
    }
    const component = shallow(<VerificationStepsModal {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('sends values on save', () => {
    const props = {
      values: ['national-id'],
      items: ['passport', 'national-id'],
      onSave: jest.fn()
    }

    const component = shallow(<VerificationStepsModal {...props} />)

    component.find('Button').prop('onClick')()
    expect(props.onSave).toHaveBeenCalledWith(props.values)
  })
})
