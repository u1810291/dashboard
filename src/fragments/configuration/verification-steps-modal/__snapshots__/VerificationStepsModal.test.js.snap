// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerificationStepModal is disaled with empty values 1`] = `
<Modal>
  <main>
    <h1>
      <FormattedMessage
        id="fragments.configuration.verification_steps_modal.title"
        values={Object {}}
      />
      <p>
        <FormattedMessage
          id="fragments.configuration.verification_steps_modal.subtitle"
          values={Object {}}
        />
      </p>
    </h1>
    <CheckboxGroup
      items={
        Array [
          Object {
            "label": <FormattedMessage
              id="verification_items.passport"
              values={Object {}}
            />,
            "value": "passport",
          },
          Object {
            "label": <FormattedMessage
              id="verification_items.national-id"
              values={Object {}}
            />,
            "value": "national-id",
          },
        ]
      }
      onChange={[Function]}
      values={Array []}
    />
  </main>
  <footer
    className="mgi-items"
  >
    <Button
      buttonStyle="primary"
      disabled={true}
      onClick={[Function]}
    >
      <FormattedMessage
        id="done"
        values={Object {}}
      />
    </Button>
  </footer>
</Modal>
`;

exports[`VerificationStepModal is enabled and has pre-checked values 1`] = `
<Modal>
  <main>
    <h1>
      <FormattedMessage
        id="fragments.configuration.verification_steps_modal.title"
        values={Object {}}
      />
      <p>
        <FormattedMessage
          id="fragments.configuration.verification_steps_modal.subtitle"
          values={Object {}}
        />
      </p>
    </h1>
    <CheckboxGroup
      items={
        Array [
          Object {
            "label": <FormattedMessage
              id="verification_items.passport"
              values={Object {}}
            />,
            "value": "passport",
          },
          Object {
            "label": <FormattedMessage
              id="verification_items.national-id"
              values={Object {}}
            />,
            "value": "national-id",
          },
        ]
      }
      onChange={[Function]}
      values={
        Array [
          "national-id",
        ]
      }
    />
  </main>
  <footer
    className="mgi-items"
  >
    <Button
      buttonStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      <FormattedMessage
        id="done"
        values={Object {}}
      />
    </Button>
  </footer>
</Modal>
`;
